### DISCUSSION TITLE
Can you please integrate A5-30-03 AFRISO Watersensor con

### DISCUSSION THREAD

**Discussion by chma73** [mak-gitdev/HA_enoceanmqtt#78] [Request]
Hi, can you please integrate A5-30-03 that would be really helpful for me. It's this device https://www.afriso.com/en/PM/Domestic-technology/AFRISO-Smart-Home/Conductivity-water-sensor-WaterSensor-con.  Thanks

**Comment by mak-gitdev** [comment]
https://github.com/users/mak-gitdev/projects/1?pane=issue&itemId=35565699

**Comment by mak-gitdev** [comment]
[Done](https://github.com/users/mak-gitdev/projects/1?pane=issue&itemId=35565699) in [0.1.27](https://github.com/mak-gitdev/HA_enoceanmqtt/commit/0d0a4cdb3d74b191feb0cc9169e11f10243d1ec9)

**Reply by chma73** [reply]
Hello, I installed 0.1.27, but I got this warning, is this normal?
2023-10-22 13:47:55,368 WARNING: Device not yet supported: A5-30-03. Only RSSI sensor will be available

**Reply by mak-gitdev** [reply]
Hello @chma73,

You may be using a custom mapping file.
Ensure that the entry `mapping_file` is empty.

If you need to use a [custom mapping file](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Examples#using-custom-mappingyaml-andor-eepxml-files), then merge your modifications into the latest one.

**Reply by chma73** [reply]
Ah you are right, I forgot to update the mapping file - thanks for your hint.

**Reply by mak-gitdev** [reply]
@chma73,

Just for the record, why do you need to have a custom mapping file ?

**Reply by chma73** [reply]
I use your mapping file, but it was not updated automatically; I'm not sure why, maybe related to the folder structure.

**Reply by mak-gitdev** [reply]
@chma73,

You should leave empty the `mapping_file` entry unless you need to customize something (it seems this is not what you want).

Please have a look at [this](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Home-Assistant-Addon#configuration) wiki page for more details.
I updated the instructions to clarify that.

**Reply by chma73** [reply]
Hm, when I follow your hint and start with an empty mapping_file I have some error messages...

2023-10-22 18:37:55,275 INFO: Mapping file correctly read: /config/e2m/enoceanmqttmapping.yaml
Traceback (most recent call last):
  File "/usr/bin/enoceanmqtt", line 33, in <module>
    sys.exit(load_entry_point('enocean-mqtt', 'console_scripts', 'enoceanmqtt')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/enocean-mqtt/enoceanmqtt/enoceanmqtt.py", line 117, in main
    com = HACommunicator(conf, sensors)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/enocean-mqtt/enoceanmqtt/overlays/homeassistant/ha_communicator.py", line 43, in __init__
    devcfg = self._ha_mapping[rorg][func][type_]['device_config']
             ~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Reply by mak-gitdev** [reply]
@chma73,

From your log, you're still using a custom file located at `/config/e2m/enoceanmqttmapping.yaml`
> 2023-10-22 18:37:55,275 INFO: Mapping file correctly read: /config/e2m/enoceanmqttmapping.yaml

Could you share your addon configuration or a screenshot of the addon configuration tab ?

In the configuration tab, the `mapping_file` entry should be kept empty.
It seems that you still set this entry to `/config/e2m/enoceanmqttmapping.yaml` but emptied that file.

**Reply by chma73** [reply]
yes of course, does it mean that the mapping file line should be empty as well?
![image](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/137674516/32fa56c3-eee2-46ca-af53-31e46dbe8e32)
![image](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/137674516/a65cca08-2683-4a63-86d4-6f5302e94626)

**Reply by mak-gitdev** [reply]
@chma73,

Yes, the `mapping_file` line shall be empty.
That's what I indicated in my previous message
>In the configuration tab, the mapping_file entry should be kept empty.

**Reply by chma73** [reply]
now I got it; it works :-), thanks for your additional supporting
